### PR TITLE
aotautograd: fix mutation bug when input is noncontiguous

### PR DESCRIPTION
**Pull request by bdhirsh**
Fixes https://github.com/pytorch/pytorch/issues/93363.

See the comment here for details: https://github.com/pytorch/pytorch/issues/93363#issuecomment-1572647261

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102767

### CODE REVIEW

File: torch/csrc/autograd/python_torch_functions_manual.cpp [hunk 1]
```diff
@@ -491,6 +496,22 @@ static PyObject* THPVariable__disable_functionalization(
...
+  static PythonArgParser parser({"_sync(Tensor t)"}, /*traceable=*/true);
+
+  ParsedArgs<1> parsed_args;
+  auto r = parser.parse(args, kwargs, parsed_args);
+  auto self_ = r.tensor(0);
+  TORCH_INTERNAL_ASSERT(at::functionalization::impl::isFunctionalTensor(self_));
+  at::functionalization::impl::sync(self_);
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
```
Comment (ezyang) [inline torch/csrc/autograd/python_torch_functions_manual.cpp:513]: Unused?

**Review by ezyang — APPROVED** [review]

File: torch/csrc/autograd/python_torch_functions_manual.cpp (hunk 1 above)
Comment (bdhirsh) [reply torch/csrc/autograd/python_torch_functions_manual.cpp:513]: `THPVariable__sync`? I think github just displayed the diff in a not-useful way. `THPVariable__sync` got renamed to `THPVariable__functionalize_has_metadata_mutation` further up, so `THPVariable__sync` got moved here.

**Pull request closed without merging**